demos/instr.h

```diff
@@ -182,38 +177,76 @@ inline void BoundsCheck(const char *str, size_t offset) {
   asm volatile(".byte 0x62, 0x02"::"a"(offset), "d"(&string_bounds):"memory");
 }
 
-// Reads an offset from the FS segment.
+// Reads an offset from the SS segment.
 // Must be inlined because the fault occurs inside and the stack pointer would
 // be shifted.
 // We fetch offset + 1 from the segment base, because the base is shifted one
 // byte below to bypass 1-byte minimal segment size.
 SAFESIDE_ALWAYS_INLINE
-inline unsigned int ReadUsingFS(unsigned int offset) {
+inline unsigned int ReadUsingSS(unsigned int ss, unsigned int offset) {
   unsigned int result;
 
   asm volatile(
-      "movzbl %%fs:(, %1, 1), %0\n"
-      :"=r"(result):"r"(offset + 1):"memory");
+      "movl %1, %%ss\n"
+      // Open a possibility for a speculation window.
```

Comment:
I'm a bit surprised this opens any speculation window -- I would expect the frontend (specifically the register renamer) to  flatten all of these to a register kill.

If we really want to wedge open the speculation window, we should look at a sequence of instructions that creates a data dependency chain. Maybe some pointer chasing across cache lines.